indico/modules/categories/controllers/display.py

```diff
@@ -167,3 +176,105 @@ def _process(self):
                            Category.chain_titles)
                  .limit(10))
         return jsonify_data(categories=[_serialize_category(c, with_path=True) for c in query], flash=False)
+
+
+class RHDisplayCategory(RHDisplayCategoryBase):
+    _category_query_options = (joinedload('children'), undefer('attachment_count'), undefer('deep_events_count'),
+                               undefer('event_count'))
+
+    def _process(self):
+        past_threshold = self.now - relativedelta(months=1, day=1, hour=0, minute=0)
+        future_threshold = self.now + relativedelta(months=1, day=1, hour=0, minute=0)
+        event_query = (Event.query.with_parent(self.category)
+                       .order_by(Event.start_dt.desc())
+                       .options(joinedload('person_links'), load_only('id', 'category_id', 'created_dt', 'end_dt',
+                                                                      'start_dt', 'title', 'protection_mode')))
+        past_event_query = event_query.filter(Event.start_dt < past_threshold)
+        future_event_query = event_query.filter(Event.start_dt >= future_threshold)
+        current_event_query = event_query.filter(Event.start_dt >= past_threshold,
+                                                 Event.start_dt < future_threshold)
+        events = current_event_query.filter(Event.start_dt < future_threshold).all()
+        events_by_month = self.group_by_month(events)
+
+        future_event_count = future_event_query.count()
+        past_event_count = past_event_query.count()
```

Comment:
Too bad SQLAlchemy doesn't seem to support `COUNT(*) FILTER (<condition>)` yet, otherwise we could easily get both counts in a single query. Anyway, doesn't make a notable difference, but useful to know that Postgres can do this